src/graphql/main/schema.graphql

```diff
@@ -230,6 +231,12 @@ type Globals {
   nodesIds: [String!]!
 }
 
+type GraphQLApplicationError implements Error {
```

Comment:
It all LGTM but I'm not sure about the semantics of `GraphQLApplicationError` replacing `InputError` and `PaymentError`. What exactly are we signifying? `ApplicationError` is very broad. Is removing the other 2 errors breaking?

Reply:
Samer and I went through this to confirm. Both errors were included as interfaces in the schema but were not actually being used anywhere. We actually pass the `IError` interface into places that need to return an error in the schema, and this was being auto-resolved to `InputError` only via `isTypeOf: () => true && ...` in the `InputError` object.

`InputError` isn't ideal for this resolution though since it has a very limited enum for the `code` field that isn't relevant. This change is to have that interface to resolve a new `GraphQLApplicationError` that has broader `code` type. I'm not sure about the name either, but the intention is that this error is for all `CustomApolloError` codes that can come through.

I had also originally considered doing a broader enum on those custom apollo errors for the `code` field ([see here](https://github.com/GaloyMoney/galoy/commit/53ea1f6205c5b6cae1bd9e359e8708087811b24c#diff-5309dec6f007762be2e567aaf4b2f973cf0a7764ca0833c4763dbdf96d524931R45)), but that broke when things like input validation return errors since those don't have codes as yet, so I reverted to simply using `GT.String` for `code`.

Reply:
p.s. should we consider adding something like this to our actions to detect breaking api changes?
https://www.the-guild.dev/graphql/inspector/docs/products/action